### PR TITLE
fix: only import webpack types

### PR DESCRIPTION
**Pull request by goloveychuk**
This PR contains:

- [ ] bugfix
- [ ] feature
- [x] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking.

List any relevant issue numbers:

### Description

So I've somehow messed with dependency tree (with monorepo+linking) and webpack loaded twise. 
While you have setup webpack as peerDep and everything should be ok, I guess better to get webpack module provided instead of importing it. Also easier ts types

### CODE REVIEW

**Comment by goloveychuk** [comment]
reapprove please

**Comment by shellscape** [comment]
Please rebase from upstream/master

File: src/index.ts [hunk 1]
```diff
@@ -93,7 +91,7 @@ class WebpackManifestPlugin implements WebpackPluginInstance {
       const hook = !NormalModule.getCompilationHooks
         ? compilation.hooks.normalModuleLoader
         : NormalModule.getCompilationHooks(compilation).loader;
-      hook.tap(hookOptions, normalModuleLoader);
+      hook.tap(hookOptions, normalModuleLoader as any);
```
Comment (shellscape) [inline src/index.ts:94]: The use of `as any` here will be a blocker for this PR

Comment (goloveychuk) [reply src/index.ts]: so without casting it won't work

Comment (shellscape) [reply src/index.ts]: look at the types before your changes (what's on master), and compare to what it is with your changes, adjust from there.

Comment (goloveychuk) [reply src/index.ts]: before my change it has import
```ts
// @ts-ignore
import NormalModule from 'webpack/lib/NormalModule';
```
so NormalModule was any. 
You suggest to replicate previous types, so I'll need to 
```ts
    const NormalModule = compiler.webpack.NormalModule as any;
```

Comment (goloveychuk) [reply src/index.ts]: I think upcasting from object to more specific type inside normalmodule hook is better, more types are preserved/checked

Comment (goloveychuk) [reply src/index.ts]: > second try. 
By this I meant that I've made second fix, removed as any.

**Comment by shellscape** [comment]
made more updates to upstream/master. webpack needed an upgrade to support node v18 in the tests.

**Comment by goloveychuk** [comment]
rebased

**Comment by goloveychuk** [comment]
I think there is no better fix, wdyt?

**Comment by wlblanchette** [comment]
Wanted to drop a comment here that this fix solves an instance of `TypeError: The 'compilation' argument must be an instance of Compilation` that results from a monorepo environment where the webpack configuration is defined in one package, with one instance of `webpack` and then imported into a second package where, for whatever reason, the second package its own instance of webpack.

In our case, it was a symptom of hoisting falling over for an unknown reason, despite the webpack versions being exactly the same. However, with a similar setup with slight differences in version, this would be reproducable.

If it's helpful to have a repro of this situation, I would be happy to spin up an issue with the repro and link it to this PR.

**Pull request merged**